src/js/components/dashboard/Dashboard.jsx

```diff
@@ -147,15 +147,17 @@ class Dashboard extends Component {
   }
 
   componentDidMount() {
-    this.config = sessionStorage.getItem('dashboardKey') || this.config;
+    const configId = this.getConfigIdFromParams();
+    this.config = configId || sessionStorage.getItem('dashboardKey') || this.config;
     if (this.props.currentLocation !== '') {
       this.fetchData(this.config);
     }
     this.loadPageFilters(this.props.activeConfig);
   }
 
   componentDidUpdate(prevProps) {
-    this.config = sessionStorage.getItem('dashboardKey') || this.config;
+    const configId = this.getConfigIdFromParams();
```

Comment:
Is this additional variable needed here as it is used only once?

Reply:
no its not